InfluxPublisher/tests/test_publisher_e2e.sh

```diff
@@ -0,0 +1,125 @@
+
+# load the influx db variables
+source ../../influxdb/influx.env
+# 
+
+# Define color codes
+GREEN='\033[0;32m'
+RED='\033[0;31m'
+NC='\033[0m' # default color
+
+#for storing output data from the influx query so we can parse. Deleted at end of run.
+output_file="influx_query_test_res.csv"
+
+sleep 6 # when triggered on an action, ensure it's been up long enought to have generated data
+
+curl -sS -X POST http://localhost:8086/api/v2/query?org=$DOCKER_INFLUXDB_INIT_ORG \
+    --output $output_file \
+    --header "Authorization: Token $DOCKER_INFLUXDB_INIT_ADMIN_TOKEN" \
+    --header "Accept: application/csv" \
+    --header "Content-type: application/vnd.flux" \
+    --data 'from(bucket:"'$DOCKER_INFLUXDB_INIT_BUCKET'")
+        |> range(start: -5s)
+        |> filter(fn: (r) => r._measurement == "xronos-dashboard-test")
+        |> aggregateWindow(every: 1s, fn: mean)
+        '
+    
+        
+# Load the CSV file and count the number of lines
+line_count=$(wc -l < $output_file)  
+
+# Check if line count is less than 20 (we have ten for both the the c and python publishers)
+if [ "$line_count" -lt 10 ]; then
+    echo -e "${RED}fail - influx db missing data  -- found " + $line_count + " lines, expected greater than 10${NC}"
+    exit 1
+else
+    echo -e "${GREEN}pass - influx db has expected data quantity.${NC}"
+fi
+
+# Find the minimum and maximum _value in the file. Min value should always be negative (since we are testing the c and python code, which generate sin and cos functions). Their averages should be opposite of each other.
+min_value=$(cut -d',' -f7 "$output_file" | tail -n +2 | sort -n | head -n 1)
+max_value=$(cut -d',' -f7 "$output_file" | tail -n +2 | sort -n | tail -n 1)
+
+# Calculate the difference
+difference=$(echo "$max_value + $min_value" | bc)
+
+# Check if the difference is within 0.01
+if (( $(echo "$difference <= 0.1" | bc -l) )); then
```

Comment:
It seems like this could flake if there is a timing irregularity in physical time that affects what data shows up in the time window. I am OK with this, but it's something we could be mindful of if any problems come up in the future

Reply:
I think timing irregularity would be a very interesting thing to detect, in which case this test failing is useful.